src/main/java/dev/cerus/xpbottler/XpBottlerPlugin.java

```diff
@@ -73,42 +74,60 @@ public void onInteract(final PlayerInteractEvent event) {
                     return;
                 }
 
+                // Assumes player always has at least one bottle because we already checked that they are holding one
+                int bottlesToFill = player.isSneaking() ?
+                    Math.min(
+                        XpBottlerPlugin.this.getPlayerExp(player) / xpPerBottle, 
+                        item.getAmount())
+                    : (XpBottlerPlugin.this.getPlayerExp(player) >= xpPerBottle) ? 1 : 0;

```

Comment:
I'm not a big fan of nested ternaries, but this is really minor so it doesn't really matter if you change it or leave it as is. :)
```suggestion
                int bottlesToFill;
                if(player.isSneaking()) {
                    bottlesToFill = Math.min(
                        XpBottlerPlugin.this.getPlayerExp(player) / xpPerBottle, 
                        item.getAmount()
                    );
                } else {
                    bottlesToFill = (XpBottlerPlugin.this.getPlayerExp(player) >= xpPerBottle) ? 1 : 0;
                }
```